requiredversiondisabler/PluginConstraintsDisabler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace RequiredVersionDisabler;
+
+use RequiredVersionDisabler\Constraints\ConstraintsCollection;
+
+class PluginConstraintsDisabler
+{
+
+    /**
+     * @var EnvironmentChecker
+     */
+    private $checker;
+    private $pluginSlug;
+    private $initFunctionName;
+
+    /**
+     * PluginConstraintsDisabler constructor.
+     *
+     * @param ConstraintsCollection $constraintsArray
+     * @param string                $pluginSlug
+     * @param string                $initFunctionName
+     */
+    public function __construct(
+        ConstraintsCollection $constraintsArray,
+        $pluginSlug,
+        $initFunctionName
+    ) {
+        $this->checker = new EnvironmentChecker(
+            $constraintsArray->constraints()
+        );
+        $this->pluginSlug = $pluginSlug;
+        $this->initFunctionName = $initFunctionName;
+    }
+
+    /**
+     * Disable the plugin if conditions apply
+     */
+    public function maybeDisable()
+    {
+        if ($this->checker->isCompatible()) {
+            return;
+        }
+        $this->disableAutomaticUpdate();
+        $this->disablePluginActivation($this->initFunctionName);
+    }
+
+    /**
+     * Disable automatic updates of this plugin
+     */
+    protected function disableAutomaticUpdate()
+    {
+        add_filter(
+            'auto_update_plugin',
+            [__CLASS__, 'notAutoUpdateThisPlugin'],
```

Comment:
This may lead to problems. Using class name here will produce a static call of the method. But this method is not static, so `PHP 7+` will produce a [warning](https://www.php.net/manual/en/language.oop5.static.php). Instead, using `$this`, like `[$this, 'notAutoUpdateThisPlugin']` will produce non-static call - exactly what we need.

There is another problem. The `notAutoUpdateThisPlugin()` method is protected, but to be used as `WP` hook callback it must be public. This is because `WP` will call it from outside the class. You can make it public or just convert this method to an anonymous function and then use it as a callback.

Sorry for the confusing explanations, feel free to ask if you need the help of having any doubts.